python/GafferImageTest/ImageMetadataTest.py

```diff
@@ -64,8 +64,8 @@ def test( self ) :
 
 		# check that we can make metadata
 
-		m["metadata"].addChild( Gaffer.NameValuePlug( "comment", IECore.StringData( "my favorite image!" ), "member1" ) )
-		m["metadata"].addChild( Gaffer.NameValuePlug( "range", IECore.V2iData( imath.V2i( 5, 10 ) ), True, "member1" ) )
+		m["metadata"].addChild( Gaffer.NameValuePlug( "comment", IECore.StringData( "my favorite image!" ), "member1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
+		m["metadata"].addChild( Gaffer.NameValuePlug( "range", IECore.V2iData( imath.V2i( 5, 10 ) ), True, "member1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
```

Comment:
Hmm, I see what you're getting at. I had originally contented myself by thinking "surely all such plugs are dynamic already, because they wouldn't serialise otherwise". I'm going to spend some time today to see if I can remove the `Dynamic` check and just propagate dirtiness for _all_ add/remove operations. My recollection is that I tried that before but it triggered a bug in construction somewhere, but maybe I can get that sorted. I'd love to consign the `Dynamic` flag to history anyway.